```
Traceback (most recent call last):
16
  File "scripts/regression_test.py", line 179, in <module>
17
    regression_test(0.15,benchmark)
18
  File "scripts/regression_test.py", line 167, in regression_test
19
    regression_status = run_h2oai(repetitions, threshold)
20
  File "scripts/regression_test.py", line 149, in run_h2oai
21
    load_h2oai(duckdb_conn,duckdb_current_conn)
22
  File "scripts/regression_test.py", line 108, in load_h2oai
23
    duckdb_current_conn.execute("CREATE TABLE x_group as SELECT * FROM 'G1_1e7_1e2_5_0.csv.gz'")
24
RuntimeError: AttributeError: 'NoneType' object has no attribute 'f_locals'
25
```

Could you look at the test failure? This seems related to the changes made in this PR.

--- a/tools/pythonpkg/src/pyconnection.cpp
+++ b/tools/pythonpkg/src/pyconnection.cpp
@@ -453,7 +453,7 @@ static unique_ptr<TableFunctionRef> ScanReplacement(const string &table_name, vo
 	auto py_table_name = py::str(table_name);
 	// Here we do an exhaustive search on the frame lineage
 	auto current_frame = py::module::import("inspect").attr("currentframe")();
-	while (current_frame) {
+	while (hasattr(current_frame, "f_locals")) {
 		auto local_dict = py::reinterpret_borrow<py::dict>(current_frame.attr("f_locals"));
 		// search local dictionary
 		if (local_dict) {

--- a/tools/pythonpkg/tests/fast/test_replacement_scan.py
+++ b/tools/pythonpkg/tests/fast/test_replacement_scan.py
@@ -1,0 +1,15 @@
+import duckdb
+import os
+
+
+class TestReplacementScan(object):
+
+	def test_csv_replacement(self, duckdb_cursor):
+		filename = os.path.join(os.path.dirname(os.path.realpath(__file__)),'data','integers.csv')
+		res = duckdb_cursor.execute("select count(*) from '%s'"%(filename))
+		assert res.fetchone()[0] == 2
+
+	def test_parquet_replacement(self, duckdb_cursor):
+		filename = os.path.join(os.path.dirname(os.path.realpath(__file__)),'data','binary_string.parquet')
+		res = duckdb_cursor.execute("select count(*) from '%s'"%(filename))
+		assert res.fetchone()[0] == 3

